bugsnag-plugin-android-anr/src/main/jni/anr_handler.c

```diff
@@ -276,40 +303,35 @@ static void install_signal_handler() {
     // We can still report to Bugsnag, so continue.
   }
 
-  if (sem_init(&reporter_thread_semaphore, 0, 0) == 0) {
+  if (sem_init(&anr_reporting_semaphore, 0, 0) == 0) {
     should_wait_for_semaphore = true;
   } else {
     BUGSNAG_LOG("Failed to init semaphore");
-    // We can still poll should_report_anr, so continue.
+    // We can still poll should_report_anr_flag, so continue.
   }
 
-  // Start the watchdog thread
+  // Start the watchdog thread sigquit_watchdog_thread_main().
   if (pthread_create(&watchdog_thread, NULL, sigquit_watchdog_thread_main,
                      NULL) != 0) {
     BUGSNAG_LOG(
         "Could not create ANR watchdog thread. ANRs won't be sent to Bugsnag.");
     return;
   }
 
-  // Install our signal handler
+  // Install our signal handler.
   struct sigaction handler;
   sigemptyset(&handler.sa_mask);
   handler.sa_sigaction = handle_sigquit;
   handler.sa_flags = SA_SIGINFO;
-  if (sigaction(SIGQUIT, &handler, &original_sigquit_handler) != 0) {
+  if (sigaction(SIGQUIT, &handler, NULL) != 0) {
```

Comment:
Added a comment here

Reply:
When bugsnag is disabled, SIGQUIT is blocked and the default handler is never called. When bugsnag is enabled, our handler gets called. Either way, the default SIGQUIT handler never gets called in an Android app (and must never be called because it would terminate the app).